File: inspect_db.py
import sqlite3
import sys

try:
    conn = sqlite3.connect('data/ICAR Collective.db')
    cursor = conn.cursor()
    
    print("--- Tables ---")
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    for table in tables:
        print(table[0])
        
    print("\n--- Schema ---")
    for table in tables:
        table_name = table[0]
        print(f"\nTable: {table_name}")
        cursor.execute(f"PRAGMA table_info('{table_name}')")
        columns = cursor.fetchall()
        for col in columns:
            print(f"  - {col[1]} ({col[2]})")
            
    conn.close()

except sqlite3.Error as e:
    print(f"An error occurred: {e}")

File: inspect_db.js
const Database = require('better-sqlite3');

try {
    const db = new Database('data/ICAR Collective.db', { readonly: true });

    console.log("--- Tables ---");
    const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table'").all();

    for (const table of tables) {
        console.log(`\nTable: ${table.name}`);
        const columns = db.prepare(`PRAGMA table_info('${table.name}')`).all();
        for (const col of columns) {
            console.log(`  - ${col.name} (${col.type})`);
        }
    }

    db.close();
} catch (err) {
    console.error("Error:", err.message);
}
